### ISSUE TITLE
Message enum in the agency spec, error handling in the provider spec

### ISSUE THREAD

**Issue by cttengsfmta** [CityOfLosAngeles/mobility-data-specification#105]
All the endpoints in the agency spec return a "message enum". The definition of this enum seems to suggest that the API returns an HTTP response with no body, and a status line with a particular status code and reason phrase.  Some of these response codes and reason phrases are non-standard. All 3xx response codes are for redirect responses, for instance. Also, some of the 2xx response phrases are non-standard. Response code 204 with phrase "No Content" is typically used for tracking cookie (sometimes called "pixel") requests, for example. See for reference:
https://www.w3.org/Protocols/rfc2616/rfc2616-sec6.html

The provider specification does not define behavior if an incorrect request is made.

We propose requiring a response with status code 400 and response phrase "Bad Request" for all requests to either API that cannot be processed. We further propose that all error responses have an [HTTP message body](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html) required to be valid JSON with the following structure

Field Name        | Type      | Required  | Defines
------------------| --------- | --------- | --------
errors            | Array     | Yes       | A list of strings. Must contain at least one error message
\-                | String    | Yes       | Informative error message

Example:
```json
{
  "errors": [
    "inconsistent coordinates specified",
    "end_time happens before start_time!"
  ]
}
```

We feel that reporting all the problems present in a particular request all at once makes it easier to troubleshoot integration problems.

**Comment by hunterowens** [comment]
@cttengsfmta This is a solid proposal for handling errors. I wonder if we want an `error` enum to supplement the string to allow for better categorization and standardization.

**Comment by hunterowens** [comment]
Agreed. I'm in the mind of "all errors have a error_catagory (enum) but the string itself can be defined by the implementor". Similar to the way python has error codes and then the error message

**Comment by cttengsfmta** [comment]
Ah, I see what you mean now. So something like

```json
{
  "errors": [
      { "category": "geo-error-enum-value",
      "message": "inconsistent coordinates specified"
      },
      {"category": "time-error-enum-value",
      "message": "end_time happens before start_time!"
      }
  ]
}
```

Sounds reasonable

**Comment by lionel-panhaleux** [comment]
Hi @hunterowens , @cttengsfmta I proposed PR 
https://github.com/CityOfLosAngeles/mobility-data-specification/pull/156 to address this issue.

Comments and insights are welcome ;-)

**Comment by hunterowens** [comment]
Hi @cttengsfmta did this issue get fully address in the proposed solution merged in #195?
